magefile.go

```diff
@@ -933,23 +930,29 @@ func packageAgent(platforms []string, packagingFn func()) {
 				"pf-host-agent",
 			}
 
-			ctx := context.Background()
+			packageRequests := []tools.PackageRequest{}
+			downloadVersion := packageVersion
+			if _, ok := os.LookupEnv(snapshotEnv); ok {
+				downloadVersion += "-SNAPSHOT"
+			}
```

Comment:
[Blocker]
Just having `SNAPSHOT` present does not mean it's true, it might as well be set to `false`. You can use `os.Getenv(snapshotEnv) == "true"` or even better, use `[strconv.ParseBool](https://pkg.go.dev/strconv#ParseBool):
```suggestion
			if ok, err := strconv.ParseBool(os.Getenv(snapshotEnv)); err==nil && ok {
				downloadVersion += "-SNAPSHOT"
			}
```

perhaps too much in one line, but you get the idea